### PR TITLE
KYC for smart contracts - Documentation

### PR DESCRIPTION
**Pull request by wsdt**
## Overview

Documentation of KYC-Use case.

## Changes

- Modified README.md.

## Testing

No code changes.

### CODE REVIEW

File: boba_community/turing-kyc/README.md [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
 
+## What is it?
+Regulation is coming, there is no doubt. Some examples: 
+
+* The famous Otherside contract from YugaLabs required KYC on launch
+* Some DeFi applications will be required to add KYC as well, depending on how regulations progress (EU, US, ..).
+
+DeFi apps currently face a huge problem when trying to comply with those new regulations. But the good news are, Turing can help with this! 
+
+With Turing you can build your own KYC gatekeeper for specific smart contract functions with ease. We have written a simple `modifier` for that, which does the check for you. All you need to do, is checking a wallet's KYC status on the AWS backend and return if the wallet is allowed to call the smart contract function.
```
Comment (InoMurko) [inline boba_community/turing-kyc/README.md:13]: Maybe expand a bit on *how* this example works. Currently it reads like a black box.

Comment (InoMurko) [reply boba_community/turing-kyc/README.md:13]: Maybe think in terms of software flows and what the user interacts with.

Comment (wsdt) [reply boba_community/turing-kyc/README.md:13]: sure! you're right

**Comment by wsdt** [comment]
added some further documentation, happy for feedback @InoMurko @souradeep-das

File: boba_community/turing-kyc/README.md [hunk 2]
```diff
@@ -1,10 +1,55 @@
...
+
+* The famous Otherside contract from YugaLabs required KYC on launch
+* Some DeFi applications will be required to add KYC as well, depending on how regulations progress (EU, US, ..).
+
+DeFi apps currently face a huge problem when trying to comply with those new regulations. But the good news are, Turing can help with this! 
+
+With Turing you can build your own KYC gatekeeper for specific smart contract functions with ease. We have written a simple `modifier` for that, which does the check for you. All you need to do, is checking a wallet's KYC status on the AWS backend and return if the wallet is allowed to call the smart contract function. 
+
+## How does it work (user perspective)?
+The idea is, that the user has to once KYC his wallet on any KYC provider that supports KYCing wallets (we are on it). The KYC process itself is completely independent of this code base, as well as the process associated with it.
```
Comment (InoMurko) [inline boba_community/turing-kyc/README.md]:
```
(we are on it)
```
I would remove this.

File: boba_community/turing-kyc/README.md [hunk 3]
```diff
@@ -1,10 +1,55 @@
...
+
+This means, that smart contracts that implement this KYC gatekeeper, will actually query the KYC status of the user's wallet and will revert for a KYC-only function, if the user hasn't KYCed his/her wallet yet. That's the magic :-). 
+
+## How does it work (developer perspective)? 
+This example basically consists just like all other Turing use cases, of a `TuringHelper.sol` and its corresponding interface called `ITuringHelper.sol`. The TuringHelper is just as usual needed, to actually be charged for your off-chain calls and to limit unauthorized access (refer to `permittedCallers` etc.). Please refer to the appendix of this ReadMe or to the general Turing documentation. 
+
+Our actual example lives in `KYCExample.sol` and its only purpose is to show you that you can constrain the access to specific smart contract functions based on whether a wallet has been KYCed (associated with a person's identity) by using a trusted KYC provider of your choice or not. As you can see, the example actually doesn't contain much code, but uses actually a so-called `modifier` named `onlyKYCed` which limits access to that function to KYCed wallets. 
+
```
Comment (InoMurko) [inline boba_community/turing-kyc/README.md:27]: I think what would be really cool is to show a sequence diagram showing the function calls during a Turing example... we get a lot of questions like that.

Comment (wsdt) [reply boba_community/turing-kyc/README.md:27]: good idea! done

File: boba_community/turing-kyc/README.md [hunk 4]
```diff
@@ -1,10 +1,58 @@
...
+description: Turing Example - OnChain KYC
 ---
 
+## What is it?
+Regulation is coming, there is no doubt. Some examples: 
+
+* The famous Otherside contract from YugaLabs required KYC on launch
+* Some DeFi applications will be required to add KYC as well, depending on how regulations progress (EU, US, ..).
+
+DeFi apps currently face a huge problem when trying to comply with those new regulations. But the good news are, Turing can help with this!
```
Comment (souradeep-das) [inline boba_community/turing-kyc/README.md]:
```suggestion
DeFi apps currently face a huge problem when trying to comply with those new regulations. But the good news is, Turing can help with this! 
```

**Review by souradeep-das — APPROVED** [review]
looks great!

File: boba_community/turing-kyc/README.md (hunk 3 above)
Comment (InoMurko) [reply boba_community/turing-kyc/README.md:27]: this is so fantastic!!!

**Pull request merged**